compiler/rustc_trait_selection/src/traits/util.rs

```diff
@@ -127,7 +127,7 @@ impl<'tcx> TraitAliasExpander<'tcx> {
         }
 
         // Get components of trait alias.
-        let predicates = tcx.implied_predicates_of(trait_ref.def_id());
+        let predicates = tcx.super_predicates_of(trait_ref.def_id());
```

Comment:
Only expand trait aliases to their super predicates, not all of their where clauses.